dotnet/src/Experimental/Process.LocalRuntime/LocalStep.cs

```diff
@@ -247,13 +241,17 @@ private async ValueTask InitializeStepAsync()
             var userStateType = genericStepType.GetGenericArguments()[0];
             if (userStateType is null)
             {
-                throw new KernelException("The generic type argument for the KernelProcessStep subclass could not be determined.");
+                var errorMessage = "The generic type argument for the KernelProcessStep subclass could not be determined.";
+                this._logger?.LogError("{ErrorMessage}", errorMessage);
```

Comment:
nit, extraneous interpolation.  The errorMessage string is still passed to logger...there's added efficiency here.